compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -397,21 +397,29 @@ object Implicits {
      *  what was expected
      */
     override def clarify(tp: Type)(implicit ctx: Context): Type = {
-      val map = new TypeMap {
-        def apply(t: Type): Type = t match {
-          case t: TypeParamRef =>
-            constraint.entry(t) match {
-              case NoType => t
-              case bounds: TypeBounds => constraint.fullBounds(t)
-              case t1 => t1
-            }
-          case t: TypeVar =>
-            t.instanceOpt.orElse(apply(t.origin))
-          case _ =>
-            mapOver(t)
+      val ctx0 = ctx
+      locally {
+        implicit val ctx = ctx0.fresh.setTyperState {
```

Comment:
It seems you can simply write
```
   implicit val ctx = ctx0.fresh.setExploreTyperState()
```
for that. The constraint is copied anyway. 

Also, it's more idiomatic to pass the new constraint to a method. I.e. instead of ctx0/locally, you could write:

```
  def map(implicit ctx: Context) = ...
  map(ctx0.fresh.setExploreTyperState())
```

Reply:
Simplified as suggested.